File: src/main/java/me/ag2s/epublib/domain/MediaType.java
package me.ag2s.epublib.domain;

import java.io.Serializable;
import java.util.Arrays;
import java.util.Collection;

public class MediaType implements Serializable {
   private static final long serialVersionUID = -7256091153727506788L;
   private final String name;
   private final String defaultExtension;
   private final Collection extensions;

   public MediaType(String name, String defaultExtension) {
      this(name, defaultExtension, new String[]{defaultExtension});
   }

   public MediaType(String name, String defaultExtension, String[] extensions) {
      this(name, defaultExtension, (Collection)Arrays.asList(extensions));
   }

   public int hashCode() {
      return this.name == null ? 0 : this.name.hashCode();
   }

   public MediaType(String name, String defaultExtension, Collection mextensions) {
      this.name = name;
      this.defaultExtension = defaultExtension;
      this.extensions = mextensions;
   }

   public String getName() {
      return this.name;
   }

   public String getDefaultExtension() {
      return this.defaultExtension;
   }

   public Collection getExtensions() {
      return this.extensions;
   }

   public boolean equals(Object otherMediaType) {
      return !(otherMediaType instanceof MediaType) ? false : this.name.equals(((MediaType)otherMediaType).getName());
   }

   public String toString() {
      return this.name;
   }
}

File: src/main/kotlin/me/ag2s/epublib/domain/MediaType.kt
package me.ag2s.epublib.domain

import java.io.Serializable

class MediaType : Serializable {
    private val name: String?
    private val defaultExtension: String?
    private val extensions: Collection<String?>
    constructor(name: String?, defaultExtension: String?) : this(name, defaultExtension, arrayOf(defaultExtension))
    constructor(name: String?, defaultExtension: String?, extensions: Array<String?>) : this(name, defaultExtension, extensions.asList())
    constructor(name: String?, defaultExtension: String?, extensions: Collection<String?>) {
        this.name = name
        this.defaultExtension = defaultExtension
        this.extensions = extensions
    }
    fun getName(): String? = name
    fun getDefaultExtension(): String? = defaultExtension
    fun getExtensions(): Collection<String?> = extensions
    override fun hashCode(): Int = name?.hashCode() ?: 0
    override fun equals(other: Any?): Boolean = other is MediaType && name!!.equals(other.getName())
    @Suppress("UNCHECKED_CAST")
    override fun toString(): String = name as String
    companion object { private const val serialVersionUID = -7256091153727506788L }
}
